ClosedXML/Excel/Cells/XLCell.cs

```diff
@@ -223,7 +223,7 @@ internal IXLCell SetValue<T>(T value, bool setTableHeader)
 
             // For SetValue<T> we set the cell value directly to the parameter
             // as opposed to the other SetValue(object value) where we parse the string and try to decude the value
-            if (value is String || value is char)
+            if (value is String || value is char || value is Guid)
```

Comment:
Now we have one branch in the code to process strings, chars, and GUIDs, and another branch to process everything else. There we have separate conditions for dates, timespans, numbers, booleans, and again for everything else. The later case converts the value using `.ToString()` and - unlike this fragment - does not care about line wrapping.

For me, it looks unclear why the particular type's processing should be added to the `SetValue` method and not to the `SetKnownTypedValue`. 
I suggest putting this logic into `SetKnownTypedValue`: add another `else if` condition for strings and chars and maybe GUIDs as well.

Reply:
String values are treated differently depending on whether you use `.SetValue(s)` or `.Value = s` .  The latter one will try to parse the value into e.g. a `DateTime`. The logic is therefore deliberately excluded from the other known types. I can shift the logic, but I will have to add a flag to `SetKnownTypeValue` to indicate whether to accept strings as they are or try to parse them into a value. Example look at this test: `SetStringCellValues`